### PR TITLE
Fix types and use buffers more consistently

### PR DESCRIPTION
**Pull request by lpsmith**

### CODE REVIEW

File: packages/hw-app-avalanche/src/Avalanche.js [hunk 1]
```diff
@@ -129,8 +136,11 @@ export default class Avalanche {
     let lastOffset = Math.floor(rawTx.length / this.MAX_APDU_SIZE) * this.MAX_APDU_SIZE;
     let lastData = rawTx.slice(lastOffset, lastOffset+this.MAX_APDU_SIZE);
     let response = await this.transport.send(0x80, this.INS_SIGN_HASH, 0x81, 0x00, lastData);
-    // TODO: This includes the 9000 at the end and the 32-byte hash at the beginning.
-    return response.toString("hex");
+    let resp = Buffer.from(response,"hex");
+    return {
+      hash: resp.slice(0,32),
```
Comment (3noch) [inline packages/hw-app-avalanche/src/Avalanche.js]: We need to assert that the response hash matches the one we gave it.

File: packages/hw-app-avalanche/src/Avalanche.js [hunk 2]
```diff
@@ -92,25 +95,29 @@ export default class Avalanche {
...
+   * @param derivation_path a path in BIP 32 format
    * @param hash hex-encoded hash to sign
-   * @return a signature as hex string
+   * @return an signature object
    * @example
    * const signature = await avalanche.signHash("44'/9000'/0'/0/0", "00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000");
    */
   async signHash(
-    path: string,
+    derivation_path: string,
```
Comment (3noch) [inline packages/hw-app-avalanche/src/Avalanche.js:113]: Even better would be for this to be a `BIPPath` probably.

File: packages/hw-app-avalanche/src/Avalanche.js [hunk 3]
```diff
@@ -137,10 +137,16 @@ export default class Avalanche {
     let lastData = rawTx.slice(lastOffset, lastOffset+this.MAX_APDU_SIZE);
     let response = await this.transport.send(0x80, this.INS_SIGN_HASH, 0x81, 0x00, lastData);
     let resp = Buffer.from(response,"hex");
```
Comment (lpsmith) [inline packages/hw-app-avalanche/src/Avalanche.js]: Why does this line even exist, again?  I mean, I thought `transport.send` returned, like, a binary buffer?   But we hex-decode it here?

Can we just get rid of this line altogether?

Comment (3noch) [reply packages/hw-app-avalanche/src/Avalanche.js]: Yeah I think you're right.

**Pull request merged**